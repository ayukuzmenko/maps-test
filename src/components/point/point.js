import React, { Component } from 'react';
import { connect } from 'react-redux';

class Point extends Component {
  deletePointHandler = () => {
    this.props.deletePoint(this.props.item.id);
  };

  render() {
    const { item } = this.props;
    return (
      <div>
        <span>{item.adress}</span>
        <button onClick={this.deletePointHandler}>x</button>
      </div>
    );
  }
}

const mapDispatch = dispatch => {
  return {
    deletePoint: dispatch.points.deletePoint,
  };
};

export default connect(
  null,
  mapDispatch,
)(Point);
