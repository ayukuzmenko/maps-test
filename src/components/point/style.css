button {
  display: inline-block;
  border: none;
  margin: 0;
  text-decoration: none;
  background: #ed0000;
  color: #ffffff;
  font-family: sans-serif;
  cursor: pointer;
  text-align: center;
  transition: background 250ms ease-in-out, transform 150ms ease;
  -webkit-appearance: none;
  -moz-appearance: none;
}

button:hover,
button:focus {
  background: #ed0000;
  outline: 1px solid #fff;
  outline-offset: -4px;
}

button:active {
  transform: scale(0.99);
}

.address {
  margin-right: 5px;
}
