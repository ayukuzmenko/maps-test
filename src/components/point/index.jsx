import React from 'react';
import { connect } from 'react-redux';
import './style.css';

const Point = props => {
  const deletePointHandler = () => {
    props.deletePoint(props.item.id);
  };

    const { item } = props;
    return (
      <React.Fragment>
        <span className="address">{item.adress}</span> 
        <button onClick={deletePointHandler}>x</button>
      </React.Fragment>
    );

}

const mapDispatch = dispatch => {
  return {
    deletePoint: dispatch.points.deletePoint,
  };
};

export default connect(
  null,
  mapDispatch,
)(Point);
