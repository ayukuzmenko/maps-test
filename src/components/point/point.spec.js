import React from 'react';
import { shallow } from 'enzyme';
import { Point } from './point';

describe('Point', () => {
  it('Add new point', () => {
    const item = { adress: 'Moscow' };
    const pointContainer = shallow(<Point item={item} />);
    expect(pointContainer).toMatchSnapshot();
    expect(
      pointContainer
        .find('span')
        .first()
        .text(),
    ).toEqual('Moscow');
  });
});
