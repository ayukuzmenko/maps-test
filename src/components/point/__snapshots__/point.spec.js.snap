// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Point Add new point 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Point
    item={
      Object {
        "adress": "Moscow",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="test"
        >
          Moscow
        </span>,
        <button
          onClick={[Function]}
        >
          x
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Moscow",
          "className": "test",
        },
        "ref": null,
        "rendered": "Moscow",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "x",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "x",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="test"
          >
            Moscow
          </span>,
          <button
            onClick={[Function]}
          >
            x
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Moscow",
            "className": "test",
          },
          "ref": null,
          "rendered": "Moscow",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "x",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "x",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
